File: src/main/java/com/nugurang/task/UserEvaluationTask.java
package com.nugurang.task;

import com.nugurang.service.UserEvaluationService;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

@Component
public class UserEvaluationTask {
    @SuppressWarnings("all")
    
    private static final org.slf4j.Logger log = org.slf4j.LoggerFactory.getLogger(UserEvaluationTask.class);
    private final UserEvaluationService userEvaluationService;

    @Scheduled(fixedDelay = 10000)
    private void evaluateUsers() {
        log.info("user evaluation task");
        userEvaluationService.evaluateUsers();
    }

    //<editor-fold defaultstate="collapsed" desc="delombok">
    @SuppressWarnings("all")
    
    public UserEvaluationTask(final UserEvaluationService userEvaluationService) {
        this.userEvaluationService = userEvaluationService;
    }
    //</editor-fold>
}

File: src/main/kotlin/com/nugurang/task/UserEvaluationTask.kt
package com.nugurang.task

import com.nugurang.service.UserEvaluationService
import com.nugurang.task.UserEvaluationTask
import io.github.oshai.kotlinlogging.KotlinLogging
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

@Component
class UserEvaluationTask(private val userEvaluationService: UserEvaluationService) {
    private val logger = KotlinLogging.logger {}
    //@Scheduled(fixedDelay = 10000)
    private fun evaluateUsers() {
        logger.info { "user evaluation task" }
        userEvaluationService.evaluateUsers()
    }
}
